### ISSUE TITLE
linter: enabling `import` plugin in override causes `no-unused-vars` violations to be reported as warnings

### ISSUE THREAD

**Issue by rtarpine-vanta** [oxc-project/oxc#21472]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

oxlint testfile.ts

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "off",
  },
  "rules": {},
  "overrides": [
    {
      "files": ["**/*.ts"],
      "plugins": ["import"],
    },
  ],
}
```

### What happened?

If I run `oxlint` with that config on this file:
```ts
const answer = 42;
```
then it warns `eslint(no-unused-vars): Variable 'answer' is declared but never used`. I didn't expect that since I thought all rules are disabled.

If I remove the `"plugins": ["import"],` line, then (as expected) no warnings or errors are reported.

I don't get any warning if I set `"plugins": ["import"]` at the top-level (outside of `overrides`), nor if I use a different plugin like `"unicorn"` in the override.
